### PR TITLE
Add new security page and update related links

### PR DESCRIPTION
**Pull request by tuhaihe**
closes: #ISSUE_Number

---

## Change logs

As the ASF site checks[1], the "Security" should link to either to a project-specifi page, or to the main http://www.apache.org/security page.

[1] https://whimsy.apache.org/pods/project/cloudberry

## Contributor's checklist

Here are some reminders before you submit your pull request:

* Make sure that your Pull Request has a clear title and commit message. You can take the [Git commit template](https://github.com/apache/cloudberry/blob/main/.gitmessage) as a reference.
* Learn the [code contribution](https://cloudberry.apache.org/contribute/code) and [doc contribution](https://cloudberry.apache.org/contribute/doc) guides for better collaboration.
* Make sure that your changes deployment preview is successful.
* List your communications in the [GitHub Issues](https://github.com/apache/cloudberry-site/issues) or [Discussions](https://github.com/apache/cloudberry/discussions) (if has or needed).
* Feel free to ask for the [cloudberry committers](https://github.com/orgs/apache/teams/cloudberry-committers) or other people to help review and approve.

### CODE REVIEW

File: src/pages/community/security.md
Comment (edespino) [inline src/pages/community/security.md:1]: How about a security page similar to this:

```
---
title: Security
description: Learn how to report security issues to us.
---

# Apache Cloudberry Security

The Apache Security Team handles all security issues across Apache projects and coordinates the response to vulnerabilities. For details on the vulnerability handling process, supported versions, and what is considered a security issue, visit: https://www.apache.org/security/

## Reporting Security Issues

DO NOT report security issues on public GitHub Issues, Jira tickets, mailing lists, or other public forums.

Send your report to: [security@apache.org](mailto:security@apache.org)

Please send one plain-text email per vulnerability with the following and additional information as necessary:
- Description of the vulnerability
- Steps to reproduce
- Affected versions
- Potential impact
- Any known mitigations
- (Optional) Suggested fix

## Public Discussion

For general security questions or discussions, please use the development mailing list: [dev@cloudberry.apache.org](mailto:dev@cloudberry.apache.org)
```

**Review by edespino — CHANGES_REQUESTED** [review]
I am recommending an update to the security page. Please let me know what you think.

File: src/pages/community/security.md
Comment (tuhaihe) [reply src/pages/community/security.md:1]: Good! I think we can take some good guides back from the old security guide. Thanks! I will update this file later.

**Comment by tuhaihe** [comment]
Hi @edespino Thank you for your draft. I have added a few more points. Could you please review it again? Thank you!

File: src/pages/community/security.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+## Do not
+
+For better collaboration, we hope you:
+
+- DO NOT report non-security-impacting bugs through this channel. If you have
+  any questions on using, development, please use [GitHub
+  Issues](https://github.com/apache/cloudberry/issues),
+  [Discussions](https://github.com/apache/cloudberry/discussions), [Dev mailing
+  list](https://github.com/apache/cloudberry/discussions/686) or
```
Comment (edespino) [inline src/pages/community/security.md:21]: The Dev mailling list link should point to the new Mailing list page.

**Review by edespino — CHANGES_REQUESTED** [review]
This PR is out of date: `This branch is out-of-date with the base branch`. Consequently, it can not reference the recently added mailling list page (https://github.com/apache/cloudberry-site/blob/main/src/pages/community/mailing-lists.md).

**Comment by tuhaihe** [comment]
> This PR is out of date: `This branch is out-of-date with the base branch`. Consequently, it can not reference the recently added mailling list page (https://github.com/apache/cloudberry-site/blob/main/src/pages/community/mailing-lists.md).

Good catch. All places have been updated. Please take a look again. Thanks!

**Review by edespino — APPROVED** [review]
+1 much better ... thank you.

**Comment by tuhaihe** [comment]
> +1 much better ... thank you.

Thanks for your feedback and review!

**Pull request merged**